qemu/tests/chardev_free_port.py

```diff
@@ -0,0 +1,47 @@
+from virttest import error_context, env_process
+from virttest import utils_misc
+
+from qemu.tests.virtio_console import add_chardev
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    qemu should try to find a free port value by to=<port> with unix socket and tcp options:
+    1) boot guest with socket with -qmp tcp:0:4444.
+    2) telnet localhost.
+    3) check output of {"execute": "qmp_capabilities"}, {"execute":"query-chardev"}.
+    4) repeat step 1 to 3 with -qmp tcp:0:4445 and telnet to port 4445.
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    chardev_infos = []
+    port_nums = []
+    vms = params.get("vms").split()
+    for vm_ind, vm in enumerate(vms):
+        _vm = env.get_vm(vm)
+        char_devices = add_chardev(_vm, params)
+        chardev_params = char_devices[0].params
+        if vm_ind == 1:
+            host = chardev_params['host']
+            chardev_to = utils_misc.find_free_ports(int(port_nums[0]) + 1, 6000, 1, host)
+            chardev_params['to'] = str(chardev_to[0])
+            chardev_params['port'] = port_nums[0]
```

Comment:
For the second vm 'vm_ind == 1', update pramrs after `add_chardev` is not correct.